fclient/federationclient.go

```diff
@@ -14,9 +15,60 @@ import (
 	"golang.org/x/crypto/ed25519"
 )
 
+// an interface for gmsl.FederationClient - contains functions called by federationapi only.
+type FederationClient interface {
+	gomatrixserverlib.KeyClient
+
+	SetInternalClient(client Client)
+	WithUserAgent(agent string)
+	DoRequestAndParseResponse(ctx context.Context, req *http.Request, result interface{}) error
```

Comment:
We'll probably need the `CreateMediaDownloadRequest` or whatever it is from `Client` being brought up to this interface. Can we rename it to `DownloadMedia` please?